### PR TITLE
Updated the Opportunity Model to match the ERD

### PR DESCRIPTION
**Pull request by GagenvirGill**
- Updated the Model
- Created a Migrations file with the changes
- Updated the types file
- Updated the Tests
- Updated the DAO methods

### CODE REVIEW

File: townhall/myapi/dao.py [hunk 1]
```diff
@@ -80,12 +81,16 @@ def filtered_opportunity(filtered_opportunity_data: FilteredOpportunityData) ->
...
-            filters['time__lte'] = filtered_opportunity_data.end_time
+        if filtered_opportunity_data.title:
+            filters['title__icontains'] = filtered_opportunity_data.title
+        if filtered_opportunity_data.starting_start_time:
+            filters['start_time__gte'] = filtered_opportunity_data.starting_start_time
+        if filtered_opportunity_data.starting_end_time:
+            filters['start_time__lte'] = filtered_opportunity_data.starting_end_time
+        if filtered_opportunity_data.ending_start_time:
+            filters['end_time__gte'] = filtered_opportunity_data.ending_start_time
+        if filtered_opportunity_data.ending_end_time:
```
Comment (ansel-rangers11) [inline townhall/myapi/dao.py:92]: Very nice

File: townhall/myapi/migrations/0007_rename_time_opportunity_start_time_and_more.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Generated by Django 5.0.6 on 2024-08-17 23:00
+
+import django.utils.timezone
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('myapi', '0006_remove_organization_description_and_more'),
```
Comment (ansel-rangers11) [inline townhall/myapi/migrations/0007_rename_time_opportunity_start_time_and_more.py:10]: Good naming of the migartion file

**Review by ansel-rangers11 — APPROVED** [review]
Looks great

**Review by RYeeAnn — APPROVED** [review]

**Comment by ansel-rangers11** [comment]
As discussed, please regenerate the migration so its 0008

**Pull request closed without merging**